### PR TITLE
Turbomole test suite

### PR DESCRIPTION
**Pull request by kunalsharma05**
Added the previously discussed implementation for Turbomole test suite. All the commits are self contained and descriptive in nature. I found making a new PR faster than actually rebasing and fixing the merge so I went ahead with this approach.
@langner @berquist @ATenderholt Please review and merge. The changes are the ones we discussed in #550.

### CODE REVIEW

**Comment by langner** [comment]
Thanks! The only thing I'd like to investigate here is the issue of file order. We dictate that order for the unit tests. There might be some questions, for example:
1) Is that the correct "canonical" order?
2) What order actually make sense
3) Should we be testing other orders as well?
4) How do we communicate this to users?

**Comment by kunalsharma05** [comment]
@langner @berquist Since we have discussed the issue of ordering the files. I think we should merge this PR.

**Comment by langner** [comment]
> @langner @berquist Since we have discussed the issue of ordering the files. I think we should merge this PR.

Yup, although there are some conflicts here now. Please resolve.

File: test/testdata [hunk 1]
```diff
@@ -199,37 +199,38 @@ Scan      ORCA         OrcaScanTest                   basicORCA3.0          dvb_
...
+SP        NWChem      NWChemKSSPTest        basicNWChem6.0              dvb_sp_ks.out
+SP        NWChem      GenericSPTest         basicNWChem6.5              dvb_sp_hf.out
+SP        NWChem      NWChemKSSPTest        basicNWChem6.5              dvb_sp_ks.out
+SP        ORCA        OrcaSPTest            basicORCA2.9                dvb_sp.out
+SP        ORCA        OrcaSPTest            basicORCA3.0                dvb_sp.out
+SP        Psi         Psi3SPTest            basicPsi3                   dvb_sp_hf.out
+SP        Psi         PsiSPTest             basicPsi4.0                 dvb_sp_rhf.out
+SP        Psi         PsiSPTest             basicPsi4.0                 dvb_sp_rks.out
+SP        QChem       GenericSPTest         basicQChem4.2               dvb_sp.out
+SP        Turbomole   GenericSPTest         basicTurbomole5.9/dvb_sp    basis         control      job.last      coord      mos
```
Comment (berquist) [inline test/testdata]: What do we think is better, this or
```
SP Turbomole GenericSPTest basicTurbomole5.9 dvb_sp/basis dvb_sp/control dvb_sp/job.last dvb_sp/coord dvb_sp/mos
```
?

Comment (kunalsharma05) [reply test/testdata]: I think dvb/filenames make it more intuitive. Will change it accordingly.

Comment (langner) [reply test/testdata]: I don't have an opinion here.

**Pull request merged**